src/k8s-engine/kubectl.js

```diff
@@ -1,32 +1,92 @@
 'use strict';
 
 const { spawn } = require('child_process');
-const os = require('os');
 const process = require('process');
 const paths = require('xdg-app-paths')({ name: 'rancher-desktop' });
+const resources = require('../resources');
+
+/**
+ *
+ * @param {Array[string]} args
+ * @param exitfunc
+ */
+function runCommand(args) {
+  return new Promise((resolve, reject) => {
+    const opts = {};
+    let output = '';
+    let errorMessage = '';
+
+    opts.env = { ...process.env };
+    opts.env.MINIKUBE_HOME = paths.data();
+
+    const kubectl = resources.executable('/bin/kubectl');
+
+    const command = spawn(kubectl, args, opts);
+
+    // TODO: For data toggle this is based on a debug mode
+    command.stdout.on('data', (data) => {
+      const buf = data.toString();
+
+      output += buf;
+      console.log(buf);
+    });
+
+    command.stderr.on('data', (data) => {
+      const buf = data.toString();
+
+      errorMessage += buf;
+      console.error(buf);
+    });
+
+    command.on('exit', (code, sig) => {
+      if (!code) {
+        resolve(output);
+      } else {
+        reject({ errorCode: code, message: errorMessage });
+      }
+    });
+  });
+}
 
 // The K8s JS library will get the current context but does not have the ability
 // to save the context. The current version of the package targets k8s 1.18 and
 // there are new config file features (e.g., proxy) that may be lost by outputting
 // the config with the library. So, we drop down to kubectl for this.
```

Comment:
We should document what `cxt` and `exitFunc` are (especially if `exitFunc` takes any arguments).

Stupid bikeshedding: [go](https://pkg.go.dev/context) usually abbreviates `context` as `ctx`, though I can see where `cxt` came from.